### ISSUE TITLE
Several build system issues

### ISSUE THREAD

**Issue by tomadamatkinson** [aws/amazon-chime-sdk-cpp#17]
Hi,

I've been attempting to use the SDK locally and have spotted several build system issues. There is a hard dependency with BoringSSL which isn't needed. Please let me know if there is a reason behind this

I also noticed that you the CMake is performing a manual combine step

```cmake
# Combine binaries
# Currently supporting Mac and Linux

set(SDK_BINARY_LIB_NAME "libamazon_chime_signaling_sdk")

if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
    set(SDK_BINARY_NAME "${SDK_BINARY_LIB_NAME}.a")

    # TODO @hokyungh: might be better to use mri script
    add_custom_command(
            TARGET ${LIBRARY_NAME}
            POST_BUILD
            COMMAND ar -x $<TARGET_FILE:${LIBRARY_NAME}>
	    COMMAND ar -x $<TARGET_FILE:protobuf::libprotobuf>
            COMMAND ar -x $<TARGET_FILE:websockets>
            COMMAND ar -qc ${SDK_BINARY_NAME} *.o
    )
elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
    set(SDK_BINARY_NAME "${SDK_BINARY_LIB_NAME}.a")
    add_custom_command(
            TARGET ${LIBRARY_NAME}
            POST_BUILD
            COMMAND
            libtool -static -o ${SDK_BINARY_NAME}
            $<TARGET_FILE:${LIBRARY_NAME}>
            $<TARGET_FILE:protobuf::libprotobuf>
            $<TARGET_FILE:websockets>
            $<TARGET_FILE:websockets_shared>
    )
endif()
```

I'm confused as to why this is being carried out as CMake builds these libraries under the hood when you use `target_link_libraries`

Is there a reason behind manually doing this step?

I am assuming that due to #16 the SDK is a few versions behind the internal repository?

Happy to discuss this further. These are a few of the issues I experienced building locally and can be solved with a few small tweaks of the cmake :smile:

**Comment by hokyungh** [comment]
If i remember correctly, `target_link_libraries` does not actually combine library. The main reason for this combining is to use it in the demo application as one library. See [here](https://github.com/aws/amazon-chime-sdk-cpp/blob/main/chime-sdk-signaling-cpp/demo/cli/cmake/FindLibSignaling.cmake#L16).

**Comment by hensmi-amazon** [comment]
The purpose of the manual combination libraries is for builders that are not using CMake (I was intending on making this an optional param soon).

I have also removed the boringSSL requirement (i still use it in the CLI tutorial because it makes building libwebrtc easier, are you using libwebrtc as well with `rtc_build_ssl=false`? Or some other libwebrtc implementation).

I know this is very late but if you are still interested feel free to rebuild and lmk if you run into any other issues.

**Comment by georgezy-amzn** [comment]
This issue has been fixed on Aug 9 with PR #29. Please reopen or submit a support case if this is still an issue.
